lib/rules/no-assignment-of-computed-property-dependencies.js

```diff
@@ -0,0 +1,196 @@
+'use strict';
+
+const emberUtils = require('../utils/ember');
+const types = require('../utils/types');
+const decoratorUtils = require('../utils/decorators');
+const javascriptUtils = require('../utils/javascript');
+const propertySetterUtils = require('../utils/property-setter');
+const assert = require('assert');
+
+const ERROR_MESSAGE =
+  "Use `set(this, 'propertyName', 'value')` or tracked properties instead of assignment for properties that are used as computed property dependencies.";
+
+/**
+ * Gets the list of string dependent keys from a computed property.
+ *
+ * @param {Node} node - the computed property node
+ * @returns {String[]} - the list of string dependent keys from this computed property
+ */
+function getComputedPropertyDependentKeys(node) {
+  if (!node.arguments) {
+    return [];
+  }
+
+  return node.arguments
+    .filter((arg) => arg.type === 'Literal' && typeof arg.value === 'string')
+    .map((node) => node.value);
+}
+
+/**
+ * Gets a list of computed property dependency keys used inside a class.
+ *
+ * @param {Node} nodeClass - Node for the Ember class
+ * @returns {String[]} - list of dependent keys used inside the class
+ */
+function findComputedPropertyDependentKeys(nodeClass) {
+  if (types.isClassDeclaration(nodeClass)) {
+    // Native JS class.
+    return javascriptUtils.flatMap(nodeClass.body.body, (node) => {
+      const computedDecorator = decoratorUtils.findDecorator(node, 'computed');
```

Comment:
This situation is _exactly_ the same as with observer dependent keys, we should probably account for that in the same rule here.

Reply:
Also, we really ought to make this check for the _imported_ computed (vs just any method that **happens** to be named `computed`).

Reply:
Updated to use imported name for `computed` and added a test.